src/frontend/src/pages/part/PartStocktakeDetail.tsx

```diff
@@ -23,6 +30,36 @@ import { TableColumn } from '../../tables/Column';
 import { InvenTreeTable } from '../../tables/InvenTreeTable';
 import { RowDeleteAction, RowEditAction } from '../../tables/RowActions';
 
+/*
+ * Render a tooltip for the chart, with correct date information
+ */
+function ChartTooltip({ label, payload }: ChartTooltipProps) {
+  if (!payload) {
+    return null;
+  }
+
+  if (label && typeof label == 'number') {
+    label = formatDate(new Date(label).toISOString());
```

Comment:
Reassigning props is an anti pattern. That's why some code quality tools suggest to wrap the props with readonly. And for such "more" resource intensiv functions it would be better to wrap them in a useMemo


```suggestion
    const formattedLabel = useMemo(() => formatDate(new Date(label).toISOString()), [label]);
```

Reply:
Yeah, great point. I have addressed this now, thanks.